### ISSUE TITLE
@rpath issue on Mac OS X Big Sur

### ISSUE THREAD

**Issue by northantech** [GrahamDumpleton/mod_wsgi#634]
I've been trying to install mod_wsgi and tried both compiling from the source and installing via pip3. In neither I was able to solve the problem. Whenever I try to restart httpd (brew installed), it throws the following error:

`httpd: Syntax error on line 185 of /usr/local/etc/httpd/httpd.conf: Cannot load lib/httpd/modules/mod_wsgi.so into server: dlopen(/usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so, 10): Library not loaded: @rpath/Python3.framework/Versions/3.8/Python3\n  Referenced from: /usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so\n  Reason: image not found`

I've got XCode, apache2.4.46 and python3.8.2 installed.

**Comment by GrahamDumpleton** [comment]
Does ``/usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so`` exist?

If it does, what do you get when you run:

```
file /usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so
```

And what do you get for:

```
file /usr/local/bin/httpd
```

Change that path to the actual path of ``httpd`` from brew if necessary.

**Comment by northantech** [comment]
mod_wsgi exists on that path and

Return values are as follows:
`file /usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so
/usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so: Mach-O 64-bit bundle x86_64`
and
`file /usr/local/bin/httpd
/usr/local/bin/httpd: Mach-O 64-bit executable x86_64`

**Comment by northantech** [comment]
Not acquainted with the background, I was able to dig into this:

`otool -l mod_wsgi.so`
`Load command 12
          cmd LC_LOAD_DYLIB
      cmdsize 104
         name /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (offset 24)
   time stamp 2 Thu Jan  1 02:00:02 1970
      current version 1770.106.0
compatibility version 150.0.0`
where 'CoreFoundation' does not exist, and it is said to be removed on Big Sur, if that is anything meaningfull.

**Comment by GrahamDumpleton** [comment]
For now I can only suggest you use ``install_name_tool`` to change the path in the ``mod_wsgi.so`` file that it uses to find the Python framework. For whatever reason, Python configuration is providing arguments to use when compiling a C extension which means that it isn't working properly.

Some details on ``install_name_tool`` in:

* https://medium.com/@donblas/fun-with-rpath-otool-and-install-name-tool-e3e41ae86172
* https://stackoverflow.com/questions/33991581/install-name-tool-to-update-a-executable-to-search-for-dylib-in-mac-os-x

So possibly something like:

```
install_name_tool --change @rpath/Python3.framework/Versions/3.8/Python3 xxx /usr/local/opt/httpd/lib/httpd/modules/mod_wsgi.so
```

where ``xxx`` is replaced with the full path name to the Python framework directory.

I don't know what ``xxx`` should be since the reference doesn't even seem to match what brew has installed for Python 3.8 for me, so not even sure if you are using brew Python or some other Python distribution.

**Comment by northantech** [comment]
I believe to have solved the problem after numerous attempts to install from source, which worked fine, but despite the `--with-python` option pointing to python3 it was compiled with python2.7 (according to `environ['SERVER_SOFTWARE']`)

Later attempts with different options, including reinstalling macOS Big Sur, did not result favorably, either, until I found out about the following from mod_wsgi documentation at [Checking Your Installation > Python Shared Libraries](https://modwsgi.readthedocs.io/en/develop/user-guides/checking-your-installation.html#python-shared-library).

> In that situation you may need to use the `--disable-framework` option to ‘configure’ script for mod_wsgi. This doesn’t actually disable use of the framework, but does change how it links to use a more traditional library style linking rather than framework linking. This seems to resolve the problems in most cases.

I think, my case was about the framework issue you mentioned, which seems to be caused by the python3.8.2 which is shipped with command-line-tools or XCode. Therefore, I installed brew python@3.9.

However, one thing to keep in mind that in Makefile, after running `./configure` and before running `make`, it might be a good idea to modify the config file name at

`LDFLAGS =  -L/usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib -L/usr/local/Cellar/python@3.9/3.9.1_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/config  -arch x86_64`

to `config-3.x-darwin` - in my case, it is config-3.9-darwin - (if critical for the installation)

Thereafter, I worked like a charm (with brew installed httpd, by the way).

The sequence of commands I used is as follows:
`./configure --disable-framework --with-python=/usr/local/bin/python3.9 --with-apxs=/usr/local/Cellar/httpd/2.4.46/bin/apxs`
Modify the Makefile as above
`sudo make && make install`
